### PR TITLE
Support shortened commit SHAs in URLs (#6450)

### PR DESCRIPTION
**Pull request by jeblair**
This supports using a git SHA of any length between 7 and 40
characters in URLs (e.g. /src/commit/SHA/...).  Previously
only commit SHAs of the full 40 character length were supported.

Signed-off-by: James E. Blair <jeblair@redhat.com>

### CODE REVIEW

**Review by stmcginnis — COMMENTED** [review]
Looks good.

File: modules/context/repo.go [hunk 1]
```diff
@@ -525,7 +525,7 @@ func getRefName(ctx *Context, pathType RepoRefType) string {
 		return getRefNameFromPath(ctx, path, ctx.Repo.GitRepo.IsTagExist)
 	case RepoRefCommit:
 		parts := strings.Split(path, "/")
-		if len(parts) > 0 && len(parts[0]) == 40 {
+		if len(parts) > 0 && len(parts[0]) >=7 && len(parts[0]) <= 40 {
```
Comment (techknowlogick) [inline modules/context/repo.go]: Issues with gofmt, the below suggestion should pass on CI
```suggestion
		if len(parts) > 0 && len(parts[0]) >= 7 && len(parts[0]) <= 40 {
```

**Comment by adelowo** [comment]
Can we support the shortened sha1 but redirect to the full actual one ?

File: modules/context/repo.go (hunk 1 above)
Comment (jeblair) [reply modules/context/repo.go]: Done, thanks!

**Comment by jeblair** [comment]
@adelowo Good question -- that also came up in the issue #6450; I responded there to try to collect that conversation in one place.

File: modules/context/repo.go [hunk 2]
```diff
@@ -491,8 +491,10 @@ const (
 	RepoRefBranch
 	// RepoRefTag tag
 	RepoRefTag
-	// RepoRefCommit commit
+	// RepoRefCommit short or long commit
```
Comment (lunny) [inline modules/context/repo.go:494]: Don't add a value in the middle of the enums.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by CirnoT** [comment]
> Can we support the shortened sha1 but redirect to the full actual one ?

We don't do that for any other endpoint that allows shortened hashes currently, so if we would want to start doing that, wouldn't it be better for another PR that updates all of them to such behavior?

**Comment by 6543** [comment]
as of my understanding still needed for https://github.com/golang/go/issues/39019 ?

**Pull request closed without merging**